syntax = "proto3";

package core_management;

// Request message for adding a function
message AddFunctionRequest {
    string function_name = 1;
    string code = 2;
    string runtime_env = 3;
    string version = 4;
}

// Response message for adding a function
message AddFunctionResponse {
    bool success = 1;
    string message = 2; // Make sure this line exists
}

// Service definition
service CoreManagement {
    rpc AddFunction(AddFunctionRequest) returns (AddFunctionResponse);
}
